pkg/platform/darwin.go

```diff
@@ -179,3 +167,108 @@ func (p PlatformDarwin) SystemAppDataDirectoryWithAutoMDM() (string, []store.Dri
 		return systemDir, []store.DriverOpt{store.WithStoreDirectory(systemDir)}
 	}
 }
+
+// GetMDMData reads data from MDM managed preferences plist
+func (p PlatformDarwin) GetMDMData() ([]byte, error) {
+	mdmPath := p.MDMConfigPath()
+	
+	// Read the plist file
+	file, err := os.Open(mdmPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open MDM plist: %w", err)
+	}
+	defer file.Close()
+	
+	// Parse the plist into a generic interface
+	var plistData any
+	decoder := plist.NewDecoder(file)
+	if err := decoder.Decode(&plistData); err != nil {
+		return nil, fmt.Errorf("failed to parse MDM plist: %w", err)
+	}
+	
+	// Convert the parsed data to JSON for consistent handling
+	jsonData, err := json.Marshal(plistData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert plist data to JSON: %w", err)
+	}
+	
+	return jsonData, nil
+}
+
+// GetMDMDataAsJSON reads MDM plist data and attempts to intelligently handle JSON strings
+// If the plist root is a JSON string, it parses and returns the JSON directly
+// If the plist contains JSON string values, it optionally expands them
+func (p PlatformDarwin) GetMDMDataAsJSON(expandJSONStrings bool) ([]byte, error) {
+	rawData, err := p.GetMDMData()
+	if err != nil {
+		return nil, err
+	}
+
+	// Try to detect if the root value is a JSON string
+	var rootValue any
+	if err := json.Unmarshal(rawData, &rootValue); err != nil {
+		return rawData, nil // Return as-is if we can't parse
+	}
+
+	// Case 1: Root value is a string that might be JSON
+	if rootStr, ok := rootValue.(string); ok {
+		// Try to parse as JSON
+		var jsonObj any
+		if err := json.Unmarshal([]byte(rootStr), &jsonObj); err == nil {
+			// It's valid JSON! Return the parsed JSON instead of the string wrapper
+			return json.Marshal(jsonObj)
+		}
+		// Not valid JSON, return the original data
+		return rawData, nil
+	}
+
+	// Case 2: Root value is a map/dict that might contain JSON strings
+	if expandJSONStrings {
+		return p.expandJSONStrings(rawData)
+	}
+
+	return rawData, nil
+}
+
+// expandJSONStrings recursively finds string values that are valid JSON and expands them
+func (p PlatformDarwin) expandJSONStrings(data []byte) ([]byte, error) {
+	var obj any
+	if err := json.Unmarshal(data, &obj); err != nil {
+		return data, nil // Return as-is if we can't parse
+	}
+
+	expanded := p.expandJSONStringsRecursive(obj)
+	return json.Marshal(expanded)
+}
+
+// expandJSONStringsRecursive recursively processes data structures to expand JSON strings
+func (p PlatformDarwin) expandJSONStringsRecursive(obj any) any {
```

Comment:
This is probably not a real security concern because the .plist file is almost certainly trustworthy, but you may still want to set a max depth here for security.